docs/keps/20200619-federated-resource-status.md

```diff
@@ -0,0 +1,537 @@
+a `Ready` condition, and otherwise report their error status.
+To do so, this property reports the status of the federated resources in their
+target clusters whenever a `ReadyCondition` is not satisfied.
+This condition would need to be identified per type or by the usage of an interface
+`IsReady` that determines this value per type of resource.
+By doing so, we ensure the `Clusters/Conditions` property shows the status of only unhealthy
+resources for the respective clusters.
+
+If we re-use the example from above and imagine a scenario where this `FederatedDeployment` resource remained `Ready=True` in two clusters, but crashed in `cluster3`.
+The value of `status.clusters.conditions` should reflect the new state for that specific cluster.
+The value of `status.conditions` defines the aggregated condition of the Federated resource itself.
+
+```yaml
+- apiVersion: types.kubefed.io/v1beta1
+  kind: FederatedDeployment
+  metadata:
+    finalizers:
+    - kubefed.io/sync-controller
+    generation: 1
+    name: asystem
+    namespace: asystem
+    resourceVersion: "70174497"
+  spec:
+    placement:
+      clusters:
+      - name: cluster2
+      - name: cluster1
+      - name: cluster3
+    template:
+      metadata:
+        labels:
+          app: nginx
+      spec:
+        replicas: 3
+        selector:
+          matchLabels:
+            app: nginx
+        template:
+          metadata:
+            labels:
+              app: nginx
+          spec:
+            containers:
+            - image: nginx
+              name: nginx
+  status:
+    clusters:
+    - name: "cluster3"
+      status: "CreationFailed"
+      conditions:
+      - lastTransitionTime: "2020-05-25T20:23:59Z"
+        lastUpdateTime: "2020-05-25T20:23:59Z"
+        status: "True"
+        type: "NotReady"
+        reason: "ReplicaFailure"
+        message: "1 Replica is not ready, you don't have enough resources"
+    conditions:
+    - lastTransitionTime: "2020-05-25T20:23:59Z"
+      lastUpdateTime: "2020-05-25T20:23:59Z"
+      status: "False"
+      type: "Propagation"
+```
+
+The value of `status.conditions` describes the propagation error that happened to `cluster3`
+while `clusters["cluster3"].conditions` contains a `NotReady` condition type with a detailed
+description of the state of the federated resource in `cluster3`.
+The value of `status.clusters["cluster3"].status` reuses the current set of available status types, e.g. `"CreationFailed"`, which identifies the current problem during the propagation.
+
+
+The status of the `FederatedDeployment` remains `Ready=True` in the rest
+of clusters: `cluster1` and `cluster2`.
+However as you can see, the `clusters[*].conditions` does not contain values with the `IsReady` condition
+for `cluster2` and `cluster1`.
+The system omits the reporting of these `Ready=True` status due two reasons:
+
+* easily decipher the propagation and other failures.
```

Comment:
My main concern is how do users know the control plane has set status correctly and completely?

If a  `status.condations` shows `Propagation = True`(as below), means both propagation and running status is ready, right?
```
status
  conditions:
  - status: "True"
    type: "Propagation"
```

Reply:
Yes, that means that the creation of the resource is considered to be ready. That could be something you might indicate with resources such as cluster roles,etc...